app/main.py

```diff
@@ -1,8 +1,39 @@
 class Person:
-    # write your code here
-    pass
+
+    people = {}
+
+    def __init__(self, name: "str", age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people.update({self.name: self})
+
+    def add_partner(self, partner_name: str) -> None:
```

Comment:
The `add_partner` method is correctly defined here, but ensure that the partner's name exists in `Person.people` before accessing it to avoid potential `KeyError` exceptions.